### ISSUE TITLE
Option to run tests directly, without spawning subprocesses (single test mode?)

### ISSUE THREAD

**Issue by Techcable** [Snaipe/Criterion#417]
Hi

I've been using your project on my Apple MacBook M1.

I've been unable to get the builtin LLDB integration to work (it crashes), so I'd like to be able to run tests "directly", without isolating them into subprocess.

That is, instead of calling `fork` to spawn a subprocess, just execute the test directly without any isolation.

This would probably work best if you restrict to running only a single test.

Lldb itself works fine with a normal process, but it's difficult to debug subprocess right now (can't follow-fork).

Thanks!

**Comment by Snaipe** [comment]
There used to be a --single flag for that express purpose but there were tons of issues around it unfortunately, see #206. If we also move forward with the cr_run proposal in #412, --single would also be fairly hard (if not reasonably impossible) to implement safely and without any weird gotchas on the user's part.

Does lldb support something like multi-inferior debugging? I'd probably look into that first, it's what I've been using with GDB for a while now since it's more flexible and lets you debug multiple failing tests concurrently.

Also, would you mind opening a new issue about the lldb integration crashing, detailing which version of criterion you used and a simple reproducer? There were some recent fixes for the M1 so this might be a good time to address them for v2.4.0.

**Comment by MrAnno** [comment]
The `--debug` option of Criterion is currently not working on macOS (M1), see #420 for more info.

As a workaround, you can debug each test case manually, for example:

```
$ ./test --debug=idle
[----] Criterion v2.4.0-rc-5-gaabfb53
[====] Running 2 tests from suite:
[----] suite::test1: Started test has PID 74869

$ sudo lldb ./test
(lldb) process attach --pid 74869
(lldb) b suite_test1_impl
(lldb) c
```

**Comment by MrAnno** [comment]
I've tried to add macOS `debugserver` support to Criterion: #421. It seems to be working in my environment if I add `debugserver` to my system `PATH`.

Give it a go, if you're interested :)

**Comment by Snaipe** [comment]
Closing in favor of #206. The original issue of --debug not working on Mac has been addressed in v2.4.0.
